### DISCUSSION TITLE
[Questions]  Messages in the dead letter queue do not expire

### DISCUSSION THREAD

**Discussion by atgtahub** [rabbitmq/rabbitmq-server#12301] [Questions] [answered]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)
- [X] I agree to provide all relevant information (versions, logs, rabbitmq-diagnostics output, detailed reproduction steps)

### RabbitMQ version used

3.12.x or older

### Erlang version used

26.0.x

### Operating system (distribution) used

CentOS

### How is RabbitMQ deployed?

RabbitMQ-as-a-Service from a public cloud provider

### Steps to deploy RabbitMQ cluster

RabbitMQ-as-a-Service from a public cloud provider

### Steps to reproduce the behavior in question

When a message reaches the dead-letter queue, it does not expire until you click get message on the rabbit management page. Is there a rabbitmq inertia setting？
Version：RabbitMQ 3.10.0 Erlang 23.2.7

**Comment by lukebakken** [comment]
You are using a version of RabbitMQ that does not receive community support from Team RabbitMQ. This is clearly stated in the community support policy.

**Comment by michaelklishin** [comment, marked as answer]
@atgtahub RabbitMQ 3.10 has been out of support (not just community support) for several years now.
If you are still using it from a public cloud provider, perhaps direct your questions to the provider you are paying, not the core team.

Messages are deleted when they reach the head of the queue. From the [TTL guide](Only when expired messages reach the head of a queue will they actually be discarded (marked for deletion)):

```
Queues that had a per-message TTL applied to them retroactively (when they already had messages)
will discard the messages when specific events occur.

Only when expired messages reach the head of a queue will they actually be discarded
(marked for deletion). Consumers will not have expired messages delivered to them.
```

In other words, in a queue that actually has consumers — a scenario RabbitMQ is designed for in many ways — those messages will be discarded and will not be delivered to consumers. In a queue that has no consumers they will keep consuming disk space until the queue is purged or deleted.
